doc improvement: doc cleanup mesh dfu (#11209)
nrfconnect/sdk-nrf: doc/nrf/protocols/bt_mesh/fota.rst

.. _ug_bt_mesh_fota:

Performing Device Firmware Updates (DFU) in Bluetooth mesh
##########################################################

The |NCS| supports two DFU methods when updating your firmware over-the-air (FOTA) for your Bluetooth mesh devices and applications.

* DFU over Bluetooth® mesh using the Zephyr Bluetooth mesh DFU subsystem that implements the Mesh Device Firmware Update Model specification version 1.0 and Mesh Binary Large Object Transfer Model specification version 1.0.

>   The specification that the Bluetooth mesh DFU subsystem is based on is not adopted yet, and therefore this feature should be used for experimental purposes only.

Reviewer: I think I'd like this to explicitly say that this means it cannot be qualified and therefore not legally used in an end product you sell.

Reviewer 2: We are using the term 'experimental' throughout the documentation. My understanding is that, it means, such feature cannot be used in production. If we start adding specifics about qualifications, we should add them at all places where experimental Bluetooth features are documented. 

What do you think?

   :local:
   :depth: 2

> In |NCS|, support for FOTA updates with point-to-point DFU over Bluetooth® Low Energy using the MCUmgr subsystem and the Simple Management Protocol (SMP) is available, in addition to support for :ref:`ble_mesh_dfu_samples`, when performing firmware over-the-air (FOTA) updates for your Bluetooth mesh devices and applications.

Reviewer 3: ```suggestion
The |NCS| supports FOTA updates with point-to-point DFU over Bluetooth® Low Energy using the MCUmgr subsystem and the Simple Management Protocol (SMP).
It also supports :ref:`ble_mesh_dfu_samples` when performing firmware over-the-air (FOTA) updates for your Bluetooth mesh devices and applications.
```

   :local:
   :depth: 2

The |NCS| supports FOTA updates with point-to-point DFU over Bluetooth® Low Energy using the MCUmgr subsystem and the Simple Management Protocol (SMP).

> It also supports :ref:`ble_mesh_dfu_samples` when performing firmware over-the-air (FOTA) updates for your Bluetooth mesh devices and applications.

Reviewer 4: We need to mention that this feature is experimental yet because it comes with mesh 1.1 specification, which is not yet finalized.

Perhaps we should add such warning to the dfu samples docs?

Author: Would adding e.g. the following statement be sufficient/correct:

Bluetooth mesh DFU support in the |NCS| is experimental.
